src/diagnostics/MOM_diagnostics.F90

```diff
@@ -290,6 +290,12 @@ subroutine calculate_diagnostic_fields(u, v, h, uh, vh, tv, ADp, CDp, p_surf, &
     if (CS%id_dv_dt>0) call post_data(CS%id_dv_dt, CS%dv_dt, CS%diag, alt_h = diag_pre_sync%h_state)
 
     if (CS%id_dh_dt>0) call post_data(CS%id_dh_dt, CS%dh_dt, CS%diag, alt_h = diag_pre_sync%h_state)
+    do k=1,nz ; do j=js,je ; do I=is-1,ie
+        ADp%du_dt(I,j,k) = CS%du_dt(I,j,k)
+    enddo ; enddo ; enddo
+    do k=1,nz ; do J=js-1,je ; do i=is,ie
```

Comment:
Same here, it seems this should be deleted since `id_hf_du_dt` is commented out in `MOM_diagnostics_init`. This look should only be exercised if `ADp%dv_dt` has been allocated.

Reply:
Same here. I need `ADp%dv_dt` to compute `dv_dt_visc_rem` in the `MOM_vert_friction` module. What do you recommend to do about allocation of `ADp%du_dt` and `ADp%dv_dt`? These variables are defined to be part of the `ADp` structure here:

https://github.com/NoraLoose/MOM6/blob/86741cd1d3b049303db325d61089263eda90979b/src/core/MOM_variables.F90#L164-L165

Reply:
I suggest allocating them in [MOM_diagnostics_init](https://github.com/NoraLoose/MOM6/blob/dev/gfdl/src/diagnostics/MOM_diagnostics.F90#L1556). 

du_dt from [diagnostics_CS](https://github.com/NoraLoose/MOM6/blob/dev/gfdl/src/diagnostics/MOM_diagnostics.F90#L70-L71) is only allocated when id_du_dt > 0, see [here](https://github.com/NoraLoose/MOM6/blob/dev/gfdl/src/diagnostics/MOM_diagnostics.F90#L1747-L1750). It also needs to be allocated when id_hf_du_dt > 0. Same for dv_dt.